### PR TITLE
Simplex hierarchical elements

### PR DESCRIPTION
**Pull request by pbrubeck**

### CODE REVIEW

File: FIAT/expansions.py [hunk 1]
```diff
@@ -24,11 +23,22 @@ def morton_index3(p, q=0, r=0):
 def jrc(a, b, n):
     """Jacobi recurrence coefficients"""
     an = (2*n+1+a+b)*(2*n+2+a+b) / (2*(n+1)*(n+1+a+b))
-    bn = (a*a-b*b) * (2*n+1+a+b) / (2*(n+1)*(2*n+a+b)*(n+1+a+b))
+    bn = (a+b)*(a-b)*(2*n+1+a+b) / (2*(n+1)*(n+1+a+b)*(2*n+a+b))
```
Comment (rckirby) [inline FIAT/expansions.py:26]: Is this for numerical accuracy?

File: FIAT/expansions.py [hunk 2]
```diff
@@ -52,10 +62,15 @@ def jacobi_factors(x, y, z, dx, dy, dz):
     return fa, fb, fc, dfa, dfb, dfc
 
 
-def dubiner_recurrence(dim, n, order, ref_pts, jacobian):
+def dubiner_recurrence(dim, n, order, ref_pts, Jinv, scale, variant=None):
```
Comment (rckirby) [inline FIAT/expansions.py:65]: A comment perhaps on the arguments and their role?  (Not that it was all that well documented before)

File: test/unit/test_fiat.py [hunk 3]
```diff
@@ -604,6 +603,68 @@ def eval_basis(f, pt):
...
+            points.extend(cell.make_points(dim, entity, degree))
+    values = B.tabulate(points)[(0,) * sd]
+    assert np.allclose(values, 0, atol=1E-12)
+
+    # test linear independence
+    m = B.get_num_members()
+    points = cell.make_points(sd, 0, degree)
+    values = B.tabulate(points)[(0,) * sd]
+    assert values.shape == (m, m)
+    assert np.linalg.matrix_rank(values.T) == m
```
Comment (rckirby) [inline test/unit/test_fiat.py:635]: Is the default tolerance (which is ~ machine eps) the correct here?  If it passes, that's a strong condition, but could we get any roundoff issue here?

**Review by rckirby — APPROVED** [review]
Just a few comments, but thanks for the contribution.

File: FIAT/expansions.py (hunk 1 above)
Comment (pbrubeck) [reply FIAT/expansions.py:26]: no, mainly aesthetics and saves one multiplication

Comment (rckirby) [reply FIAT/expansions.py:26]: When a and b are close together, then a^2-b^2 is probably better than (a+b)(a-b) if a, b are large but worse if they are small?  At any rate, a = b + \epsilon is not a use case as far as I'm aware?

Comment (pbrubeck) [reply FIAT/expansions.py:26]: They're both integers, moreover b \in {0, 1}

**Pull request merged**